dnsprovider/aws.go

```diff
@@ -46,6 +47,26 @@ type AWSProvider struct {
 	DryRun bool
 }
 
+// NewAWSProvider initializes a new AWS Route53 based DNSProvider.
+func NewAWSProvider(dryRun bool) (DNSProvider, error) {
+	config := aws.NewConfig()
+
+	session, err := session.NewSessionWithOptions(session.Options{
```

Comment:
In another project we wanted to have `SharedConfigEnable` so it detects the region better. Do you know a better way?